Statistikk controller

torkelhi opened the pull request:
StatistikkController.cs
 - Ulike GET forspørsler.

Statistikk.cs 
- Model for Statistikk

Commit:
Innit commit på StatistikkController.

4 metoder er blitt implementert.
Metoder for å hente ut antall forslag laget av bruker eller team med filtrering av status.

Commit:
Uke - Måned handlers

Lagt til handlers for å hente ut forslag relatert til bruker/team og uke/måned.

Commit:
Ny model og handlers for ukentlig/måndtlig statistikk ferdigstilt.

Små endringer i SQL kode ved samtlige handlers.
Lagt til en model (Statistikk.cs) som skal brukes til razor page.

torkelhi commented:
Jeg mener man kan resolve konflikten

menymo commented:
@torkelhi Konflikten er bare en newline?

torkelhi commented:
Ja jeg tror det.

menymo commented on PDSA_System/Server/Models/Statistikk.cs:
```diff
@@ -0,0 +1,19 @@
+
+        public string Status { get; set; }
+
+        public int Stat { get; set; }
```
Kunne det vært mulig å få nærmere dokumentasjon på hva dette atributtet betyr? Helst burde det hete `Count` dersom det alltid vil peke mot et antall

```suggestion
        public int Count { get; set; }
```

torkelhi replied:
Takk. Årsaken til at jeg ikke bruker Count som attributt er fordi Count er et reservertnøkkel ord i SQL språket. Så frykter at Count(*) AS Count kan forårsake feil.

Stat var ment som forkortelse for statistikk.

menymo replied:
Jeg gjorde et forsøk på det, og det ser ut til å fungere

menymo reviewed (changes requested):
Dette var fantastisk, alt funket på første forsøk :)

Ligger et lite forslag angående navngiving, men ellers helt perfekt

menymo committed:
Merge branch 'master' into statistikk-controller

menymo commented:
Når jeg tenker meg om, så burde det også være en endpoint for å finne statistikk mellom to timestamps, slik at en leder kan selv generere kvartalsrapperter eller noe sånt. Det er noe som kan implementeres senere, men er definitivt på "should have"

Commit:
Endringer etter review

Stat --> Count

Commit:
Optimalisering av SQL.

var en bug hvor differansen var -X og dermed returnerte true når den egentlige skulle være false.
Den er nå fikset.

Commit:
Forslag statistikk GROUP BY TeamId.

Metode som fører statistikk på antall forslag per team.

Commit:
Brukerdefinert tidsperiode

To nye metoder, en metode for bruker og en for team.
Henter statistikk på antall forslag for en spesifikk bruker eller team innenfor en brukerdefinert tidsperiode.

Commit:
GetStatistikkBetween

Lagt til en metode som henter ut alle forslag innenfor en brukerdefinert tidsperiode.

torkelhi committed:
Merge branch 'master' into statistikk-controller

menymo reviewed (approved):
Funka bra, tror det blir ganske greit å jobbe med denne APIen i frontend
